### PR TITLE
Fix #34

### LINKED ISSUE #34: give user feedback about hashdb background thread
**Problem Statement**
Give the user some indication that there is a hashdb thread running in the background.

Currently IDA will freeze up when adding a large enum which can cause a poor user experience if the user forgets that there is a background thread running.

**Proposed Solution**
IDA is limited in the type of feedback we can give the user so the current idea is to temporarily add a toolbar when the thread is active and remove it once the thread completes (or crashes). The toolbar could also be used to kill the thread if the user wants.

[Brutal IDA](https://github.com/tmr232/BRUTAL-IDA) has template code that could re-used.

### PR DESCRIPTION
**Pull request by anthonyprintup**
This PR fixes issue #34.

## Detailed description:
- added a new import `ida_typeinf` for typeinf related APIs,
- added a new helper function `get_existing_enum_values` which checks if an enum already exists and parses its values using the IDA API ([ida_typeinf.get_named_type](https://www.hex-rays.com/products/ida/support/idapython_docs/ida_typeinf.html#ida_typeinf.get_named_type), [ida_typeinf.idc_print_type](https://www.hex-rays.com/products/ida/support/idapython_docs/ida_typeinf.html#ida_typeinf.idc_print_type)),
- `add_enums` will now check for existing enum values before parsing the provided hash list,
- `add_enums` now manually corrects strings with dots in their names (e.g. `ntdll.dll` -> `ntdll_dll`),
- `add_enums` now uses `ida_name.get_name_ea` to check if a name already exists in the database (previously used `ida_enum.add_enum_member` which checked this automatically)
- `add_enums` no longer expects a 3rd function parameter because the enum size now depends on the database type or the algorithm size (64-bit databases will always generate 64-bit enums),
- `add_enums`: replaced the method used when creating an enum (massive performance improvements); using [ida_typeinf.idc_parse_types](https://www.hex-rays.com/products/ida/support/idapython_docs/ida_typeinf.html#ida_typeinf.idc_parse_types) and [ida_typeinf.import_type](https://www.hex-rays.com/products/ida/support/idapython_docs/ida_typeinf.html#ida_typeinf.import_type).

## Possible (but unlikely) bugs:
- `get_existing_enum_values` will not parse values properly on expected integer bases (e.g. if for some reason the values aren't in decimal or hexadecimal form),
- because we no longer use `ida_enum.add_enum_member` to append enum members we lose some internal corrections that IDA performs on the enum names (e.g. replacing dots with underscores); this has to be corrected manually from now on.

## Disclaimer:
I've only tested the plugin on two samples (one 32-bit and one 64-bit). Strings with invalid names (e.g. `-path`) haven't been tested yet, but should work (look at #31).
TLDR; more testing required before we can push a release.

### CODE REVIEW

**Comment by anthonyprintup** [comment]
Added support for IDA 8.0 by using `ida_enum.get_enum_member_by_name` to check for enum names.

**Comment by anthonyprintup** [comment]
> I've only tested the plugin on two samples (one 32-bit and one 64-bit). Strings with invalid names (e.g. -path) haven't been tested yet, but should work (look at https://github.com/OALabs/hashdb-ida/pull/31).
Confirmed to be working properly in all instances.

All tests passed (IDA 8.0, Python 3.10.6) and ready to merge.
